client/src/App.js

```diff
@@ -27,61 +31,109 @@ const App = () => {
   const [itemList, dispatch] = useReducer(reducer, []);
   const [cart, setCart] = useState([]);
   const [isAuth, setIsAuth] = useState(false);
-  const [user, setUser] = useState({});
+  const [user, setUser] = useState({
+    name: { firstName: "", lastName: "" },
+    email: "",
+    phone: "",
+  });
+  const [loaded, setLoaded] = useState(false);
+  const [admin, setAdmin] = useState(false);
   //side effect when page first time rendered
   useEffect(() => {
-    const URL = "/fetchItems";
-
-    const fetchJSON_fromServer = async () => {
-      const itemList = await (await fetch(URL)).json();
-      dispatch({ type: "fetchedItemJson", payload: itemList });
+    const fetchItems_fetchUser = async () => {
+      //fetching user data from the server side if any
       const userData = await (await fetch("/getUserDetails")).json();
```

Comment:
don't use nested await. Either use await in different vars or use then-catch statements